From 511e892d496980fae6a756eba497d745e596a68f Mon Sep 17 00:00:00 2001
From: Xuefer <xuefer@gmail.com>
Date: Sat, 4 Jul 2015 12:26:56 +0800
Subject: [PATCH 1/3] fix segv on const string

---
 service/service.c | 6 ++++--
 service/service.h | 2 +-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/service/service.c b/service/service.c
index 2702dc3..ca82538 100644
--- a/service/service.c
+++ b/service/service.c
@@ -508,17 +508,19 @@ static struct ubus_object main_object = {
 };
 
 int
-service_start_early(char *name, char *cmdline)
+service_start_early(const char *name, const char *cmdline)
 {
 	void *instances, *instance, *command, *respawn;
+	char *cmdlinecopy = alloca(strlen(cmdline) + 1);
 	char *t;
+	strcpy(cmdlinecopy, cmdline);
 
 	blob_buf_init(&b, 0);
 	blobmsg_add_string(&b, "name", name);
 	instances = blobmsg_open_table(&b, "instances");
 	instance = blobmsg_open_table(&b, "instance1");
 	command = blobmsg_open_array(&b, "command");
-	t = strtok(cmdline, " ");
+	t = strtok(cmdlinecopy, " ");
 	while (t) {
 		blobmsg_add_string(&b, NULL, t);
 		t = strtok(NULL, " ");
diff --git a/service/service.h b/service/service.h
index 892a147..28ffc1c 100644
--- a/service/service.h
+++ b/service/service.h
@@ -49,7 +49,7 @@ struct service {
 void service_validate_add(struct service *s, struct blob_attr *attr);
 void service_validate_dump(struct blob_buf *b, struct service *s);
 void service_validate_dump_all(struct blob_buf *b, char *p, char *s);
-int service_start_early(char *name, char *cmdline);
+int service_start_early(const char *name, const char *cmdline);
 void service_validate_del(struct service *s);
 void service_validate_init(void);
 void service_init(void);
-- 
2.4.10

